text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,600 @@
+
+`forceAliasCreation` can be used only when the user pool is configured with either email or phone as a verified contact
+method. When this is set, any existing user with the same email address or phone number will have its alias migrated to
+this user.
+
+When the `resendInvitation` property is set to true, Cognito will resend the invitation message for a user that already
+exists but has not signed up.
+
+Users can be created for imported user pools by using the `UserPoolUser` construct, as so -
+
+```ts
+const userpool = UserPool.fromUserPoolAttributes(this, 'myuserpool', { ... });
+
+new UserPoolUser(this, userpool, {
+  userPool: userpool,
+  // ...
+  // all of the same properties as above
+  // ...
+});
+```
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * deliveryMedium: ContactsMethod.PHONE
+> * attributes: none
+> * forceAliasCreation: false
+> * resendInvitation: false
+
+### Groups
+
+User pools can be configured with groups into which users are added. Groups is the primary mechanism in Cognito for
+permission assignment.
+
+Add a new group to the user pool using the `createGroup()` API, as so -
+
+```ts
+const userpool = new UserPool(this, 'myuserpool', { ... });
+
+const customerrole = new iam.Role(this, 'customer-role', { ... });
+
+userpool.createGroup(this, 'myuserpool-customers', {
+  groupName: 'customers',
+  description: 'group for myawesomeapp customers',
+  role: customerrole
+});
+```
+
+The required `groupName` and the optional `description` properties are the identifiers for this group.
+
+The `role` property refers to the IAM role whose permissions that the users in this group are able to assume.
+
+When several different groups are defined for the same user pool, they can be ordered to have a precendence. Precedence
+will determine which IAM role's permission will be available for a user when they are part of two different groups. See
+[CreateGroup
+API](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateGroup.html#CognitoUserPools-CreateGroup-request-Precedence)
+for more details.
+
+Use the `takesPrecendenceOver()` API to set the precedence of one group over the other.
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * description - empty
+> * precedence - `undefined`
+
+## Identity Pools (Federated Identity)
+
+Identity pools enables creation of unique identities for their users and federate them with identity providers. These
+identity providers can be a Cognito user pool, Facebook, Google, or any other SAML-based identity provider.
+With an identity pool, you can obtain temporary, limited-privilege AWS credentials to access other AWS services.
+
+Using the CDK, a new identity pool can be created as part of the stack using the construct's constructor. You must
+specify the `identityPoolName`. This is the quickest way to create an identity pool using all of the defaults -
```

Comment:
same thought as earlier comment: are we trying to be deliberate about avoiding CFN generated names and therefore made this a required property?